domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,318 @@
+package org.oppia.android.domain.classify.rules.fractioninput
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.Fraction
+import org.oppia.android.app.model.InteractionObject
+import org.oppia.android.domain.classify.InteractionObjectTestBuilder
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputIsEquivalentToRuleClassifierProviderTest {
+
+  private val WHOLE_NUMBER_VALUE_123 =
+    InteractionObjectTestBuilder.createWholeNumber(isNegative = false, value = 123)
+  private val WHOLE_NUMBER_VALUE_254 =
+    InteractionObjectTestBuilder.createWholeNumber(isNegative = false, value = 254)
+  private val FRACTION_VALUE_2_OVER_8 =
+    InteractionObjectTestBuilder.createFraction(isNegative = false, numerator = 2, denominator = 8)
+  private val FRACTION_VALUE_1_OVER_5 =
+    InteractionObjectTestBuilder.createFraction(isNegative = false, numerator = 1, denominator = 5)
+  private val FRACTION_VALUE_33_OVER_2 = InteractionObjectTestBuilder.createFraction(
```

Comment:
`FRACTION_VALUE_33_OVER_2 ` but the numerator is `133`

Reply:
Done